File: 38-count-and-say/38-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string num = "1";
        
        while(n-->1){
            string current = num+" ";
            char prev = current[0];
            int i=0;
            int count=0;
            string temp = "";
            while(i<current.size()){
                if(current[i]==prev){
                    count++;
                }
                else{
                    temp += to_string(count);
                    temp += prev;
                    count=1;
                }
                prev = current[i];
                i++;
                
            }
            num = temp;
        }
        
        return num;
    }
};

File: 38-count-and-say/38-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n==1: return "1"
        base = self.countAndSay(n-1)
        cur,r,cnt,N,ans = None,0,0,len(base),""
        while r<N:
            cur = base[r]
            while r<N and base[r]==cur:
                r+=1
                cnt+=1
            print(cur,cnt)
            ans+=str(cnt)+cur
            cnt=0
            
        return ans
            
            